src/Symfony/Bridge/Twig/Validator/Constraints/TwigValidator.php

```diff
@@ -45,37 +45,43 @@ public function validate(mixed $value, Constraint $constraint): void
 
         $value = (string) $value;
 
-        if (!$constraint->skipDeprecations) {
-            $prevErrorHandler = set_error_handler(static function ($level, $message, $file, $line) use (&$prevErrorHandler) {
-                if (\E_USER_DEPRECATED !== $level) {
-                    return $prevErrorHandler ? $prevErrorHandler($level, $message, $file, $line) : false;
-                }
-
-                $templateLine = 0;
-                if (preg_match('/ at line (\d+)[ .]/', $message, $matches)) {
-                    $templateLine = $matches[1];
-                }
-
-                throw new Error($message, $templateLine);
-            });
-        }
-
         $realLoader = $this->twig->getLoader();
         try {
             $temporaryLoader = new ArrayLoader([$value]);
             $this->twig->setLoader($temporaryLoader);
+
+            if (!$constraint->skipDeprecations) {
+                $prevErrorHandler = set_error_handler(static function ($level, $message, $file, $line) use (&$prevErrorHandler) {
+                    if (\E_USER_DEPRECATED !== $level) {
+                        return $prevErrorHandler ? $prevErrorHandler($level, $message, $file, $line) : false;
+                    }
+
+                    $templateLine = 0;
+                    if (preg_match('/ at line (\d+)[ .]/', $message, $matches)) {
+                        $templateLine = $matches[1];
+                    }
+
+                    throw new Error($message, $templateLine);
+                });
+            }
+
             $this->twig->parse($this->twig->tokenize(new Source($value, '')));
+
+            if (!$constraint->skipDeprecations) {
+                restore_error_handler();
+            }
         } catch (Error $e) {
+            if (!$constraint->skipDeprecations) {
+                restore_error_handler();
```

Comment:
This won't restore the error handler for other errors. You need to use a `try/finally` block for that (a nested one then)

Reply:
oh indeed, should be better now